test/distributed/org/apache/cassandra/distributed/test/sai/IndexDDLTest.java

```diff
@@ -0,0 +1,557 @@
+    {
+        String query = String.format("SELECT ID1 FROM %s.%s WHERE V2='0'", KEYSPACE, table);
+
+        Object[][] rows = level == null
+                          ? cluster.get(node).executeInternal(query)
+                          : cluster.coordinator(node).execute(query, level);
+
+        Assert.assertEquals(stringIndexRows, rows.length);
+    }
+
+    private int repair(int node)
+    {
+        return cluster.get(node).callOnInstance(() -> {
+            SimpleCondition repairFinished = new SimpleCondition();
+
+            StorageService.instance.addNotificationListener((notification, handback) ->
+            {
+                // TODO: Find a cleaner way to determine that this is a completion notification.
+                if (notification.getMessage().toLowerCase().contains("repair") && notification.getMessage().toLowerCase().contains("finished"))
+                {
+                    repairFinished.signalAll();
+                }
+            }, null, null);
+
+            RepairOption options = new RepairOption(RepairParallelism.SEQUENTIAL, false, false, false, 1, Collections.emptyList(), false, false, false, PreviewKind.NONE, false);
+            int ranges = StorageService.instance.repairAsync(KEYSPACE, options.asMap());
+
+            if (ranges < 1)
+            {
+                return ranges;
+            }
+
+            // If the repair doesn't finish, something has gone terribly wrong...
+            try
+            {
+                if (repairFinished.await(5, TimeUnit.MINUTES))
+                    return ranges;
+            }
+            catch (Throwable e)
+            {
+                throw Throwables.unchecked(e);
+            }
+            throw new IllegalStateException("Repair did not finish!");
+        });
+    }
+
+    private void decommission(int node)
+    {
+        cluster.get(node).runOnInstance(() -> {
+            try
+            {
+                StorageService.instance.decommission(false);
+            }
+            catch (Throwable e)
+            {
+                // expected to have IllegalStateException becuase in-jvm dtest doesn't register a daemon
+                if (e instanceof IllegalStateException && e.getMessage().contains("No configured daemon"))
+                    return;
+                throw Throwables.unchecked(e);
+            }
+        });
+    }
+
+    public void cleanup(int node, String keyspace, String table)
+    {
+        cluster.get(node).runOnInstance(() -> {
+            try
+            {
+                StorageService.instance.forceKeyspaceCleanup(keyspace, table);
+            }
+            catch (Throwable e)
+            {
+                throw Throwables.unchecked(e);
+            }
+        });
+    }
+
+    /**
+     * Run repeated verification task concurrently with target test
+     */
+    protected static class TestWithConcurrentVerification
```

Comment:
I think that all these concurrent helpers can be replaced with Awaitality.
https://github.com/awaitility/awaitility/wiki/Usage
This library is already present in OSS C* tunk

Reply:
good idea, I will rebase `storage_attached_index` branch with latest trunk..

Reply:
updated with `Awaitility`..